### ISSUE TITLE
Dyson Sphere Program (1366540)

### ISSUE THREAD

**Issue by Fatmice** [ValveSoftware/Proton#4572]
# Compatibility Report
- Name of the game with compatibility issues: Dyson Sphere Program
- Steam AppID of the game: 1366540

## System Information
- GPU: Radeon VII
- Driver/LLVM version: Mesa 20.2.3/10.0.1
- Kernel version: 5.10.8-100.fc32.x86_64
- Link to full system information report as: [system-information.log](https://github.com/ValveSoftware/Proton/files/5880512/system-information.log)
- Proton version: 5.13-5 ; DXVK 1.7.3-36-g85c70ad5/1.2.131

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

[steam-1366540.log](https://github.com/ValveSoftware/Proton/files/5880479/steam-1366540.log)

## Symptoms 
Middle mouse can not be used to rotate the camera

## Reproduction
Launch game with compatibility set for Proton 5.13-5.  Load up save and hold middle mouse to try to rotate the camera. No response.  At the moment, key rebind is not yet a feature.  The game seems to be made with Unity version 2018.4.12f1

**Comment by cinnaboot** [comment]
I had a similar input problem that seems to be helped by disabling the in-game overlay in the steam game options.

**Comment by Fatmice** [comment]
> I had a similar input problem that seems to be helped by disabling the in-game overlay in the steam game options.

Ah, I just tried that.  No change to the functionality of the middle mouse button in my case.

**Comment by Fatmice** [comment]
I should mention that I'm using a Logitech G600 mouse.  I do not know if the many buttons on it casues issue for this particular game.

**Comment by dtg01100** [comment]
On my machine, setting gnome to use the xorg session worked around the problem. Tested with both the rpmfusion and flatpak packaged versions of steam.

**Comment by Fatmice** [comment]
> On my machine, setting gnome to use the xorg session worked around the problem. Tested with both the rpmfusion and flatpak packaged versions of steam.

Yeah, I fully suspect it's a wayland issue of some sort.  But I want to stick with gnome on wayland as X is deprecated

**Comment by dtg01100** [comment]
Same for me, also I have tried the wine virtual desktop, didn't work either. Oddly, middle click drag does work in the tech tree, so there are parts grabbing the event.

**Comment by Fatmice** [comment]
> Same for me, also I have tried the wine virtual desktop, didn't work either. Oddly, middle click drag does work in the tech tree, so there are parts grabbing the event.

Yes, this is true.  The middle mouse button does work while the tech tree screen is open.  So clearly some event is being missed when you are under 3D rendering.

**Comment by Fatmice** [comment]
> Same for me, also I have tried the wine virtual desktop, didn't work either. Oddly, middle click drag does work in the tech tree, so there are parts grabbing the event.

Do you think this should be mentioned on gnome issue tracker?

**Comment by dtg01100** [comment]
Perhaps, at least to get the issue on their radar while the game still has attention on it.

**Comment by Fatmice** [comment]
I've made issue on gnome-mutter.  I hope it is the right place.
https://gitlab.gnome.org/GNOME/mutter/-/issues/1628

**Comment by Fatmice** [comment]
A little update, Olivier Fourdan tried to help with a test build of XWayland but that did not improve the middle-mouse function.  It appears that "it's indeed unlikely a problem with pointer warp."  There is a scratch build that's still available here https://kojipkgs.fedoraproject.org//work/tasks/8068/61188068/.  If anyone with the issue wants to test it, download from there and read the previous thread on how to use.

It would help if people with hardware capable of running this game run it under Wayland to help narrow down the issue.

**Comment by Fatmice** [comment]
Olivier has determined that problem lies with XWayland.
https://gitlab.freedesktop.org/xorg/xserver/-/issues/1130

**Comment by Fatmice** [comment]
@dtg01100 
New scratch build for testing (not for using)
https://koji.fedoraproject.org/koji/taskinfo?taskID=61720323

**Comment by PraetorianSC4** [comment]
Game runs pretty well under proton, with a minor exception of memory maxing out at 1.5GB before the game starts lagging. This requires a restart, and then its a steady crawl  back up to that point. Reasons for this are unclear (for me). Anyone else notice this?

**Comment by samip5** [comment]
I'm unable to load a save file at all, even if it's a completely new or one used in Windows.

Running Gentoo, kernel 5.14, mesa 21.1.7 (vulkan USE flag enabled).

System information and logs: gist [here](https://gist.github.com/samip5/5a47cbbaf6299094864bb18198f387bc)

Proton versions tested: Experimental and 6.3-6.

Screenshot of the in-game error: 
![image](https://user-images.githubusercontent.com/1703002/132144810-27afcf9c-0bf0-4ebf-a403-071d9a5afb27.png)

**Comment by kisak-valve** [comment]
Hello @samip5, `eventfd: Too many open files` looks like the key detail in your log. Please check [`ulimit -Hn`](https://github.com/ValveSoftware/Proton/wiki/Requirements#fd-limit-requirements-systemd--240) and verify it gives you a high value and not 4096.

**Comment by samip5** [comment]
> Hello @samip5, `eventfd: Too many open files` looks like the key detail in your log. Please check [`ulimit -Hn`](https://github.com/ValveSoftware/Proton/wiki/Requirements#fd-limit-requirements-systemd--240) and verify it gives you a high value and not 4096.

Yeah, it says 4096..  So that's more than likely the issue.

**Comment by samip5** [comment]
> > Hello @samip5, `eventfd: Too many open files` looks like the key detail in your log. Please check [`ulimit -Hn`](https://github.com/ValveSoftware/Proton/wiki/Requirements#fd-limit-requirements-systemd--240) and verify it gives you a high value and not 4096.
> 
> Yeah, it says 4096.. So that's more than likely the issue.

I can verify that increasing it fixed the issue.

**Comment by Chais** [comment]
I'm not sure if I'm dealing with the same issue. So if I should open a new issue, please let me know.
I _can_ rotate the camera with middle click, but it misbehaves.  
- The rotation speed depends on how far my cursor is from the left edge of the screen. The further, the faster. Distance from top or bottom edge does not seem to play a role.
- I can only rotate the camera clockwise (image rotates counter-clockwise) no matter how I move the cursor.
- Camera invariably rotates into a birds-eye view on my mecha.

## System Information
- GPU: VEGA 64
- Driver/LLVM version: Mesa 21.2.5/13.0.0
- Kernel version: 5.15.2-zen1-1-zen
- Proton version: 6.3-7 ; DXVK 1.9.2

**Comment by radgeRayden** [comment]
Game has very slow FPS. I have a ryzen 9 5900X and a RTX3080. At 1080p I get 20fps.

PS: turns out I was low on VRAM. The real bug here is steamwebhelper hogging 4GB.

**Comment by tbe** [comment]
I also see bad FPS on a Threadripper 2970WX + 6700XT. Neither CPU nor GPU are at full load. There also is no single CPU core that has full load.

radeontop shows also no obvious bottle neck. The last time i played (before the Rise of the dark fog update) it worked perfectly fine.
